### ISSUE TITLE
Scale incorrect in atlas.urdf files

### ISSUE THREAD

**Issue by osrf-migration** [osrf/drcsim#275]
**Original report ([archived issue](https://osrf-migration.github.io/drcsim-gh-pages/#!/osrf/drcsim/issues/275)) by Jesper Smith (Bitbucket: [jespersmith](https://bitbucket.org/%7B61d9fe7d-e554-408c-99fb-faefa9c96114%7D/)).**

The original report had attachments: [NewModelImport.png](https://osrf-migration.github.io/drcsim-gh-pages/data/repositories/osrf/drcsim/issues/275/attachments/NewModelImport.png), [models.zip](https://osrf-migration.github.io/drcsim-gh-pages/data/repositories/osrf/drcsim/issues/275/attachments/models.zip), [atlas_simple_shapes.urdf](https://osrf-migration.github.io/drcsim-gh-pages/data/repositories/osrf/drcsim/issues/275/attachments/atlas_simple_shapes.urdf)

----------------------------------------
In atlas.urdf, the scale for several links is set to 100.0. Looking at the corresponding .dae file the resolution of the joint is set to centimeters

    <unit meter="1.0" name="centimeter"></unit>

This screws up our model loader, as it uses the correct units and now the links are extremely big. Especially considering .sdf/.urdf are interchange formats, it is poor form to compensate for errors in the model file (I can disable scaling in our loader but everything that actually uses scale will be screwed up).

**Comment by osrf-migration** [comment]
**Original comment by Steve Peters (Bitbucket: [Steven Peters](https://bitbucket.org/%7B2ccfed09-18b8-4921-8d58-15ef01092802%7D/), GitHub: [scpeters](https://github.com/scpeters)).**

----------------------------------------
It looks like the following are the affected meshes:
~~~
<mesh filename="package://atlas/meshes/l_lleg.dae" scale="100. 100. 100." />
<mesh filename="package://atlas/meshes/l_uleg.dae" scale="100. 100. 100." />
<mesh filename="package://atlas/meshes/pelvis.dae" scale="100. 100. 100." />
<mesh filename="package://atlas/meshes/r_larm.dae" scale="100. 100. 100." />
<mesh filename="package://atlas/meshes/r_lglut.dae" scale="100. 100. 100." />
<mesh filename="package://atlas/meshes/r_lleg.dae" scale="100. 100. 100." />
<mesh filename="package://atlas/meshes/r_uarm.dae" scale="100. 100. 100." />
<mesh filename="package://atlas/meshes/r_uglut.dae" scale="100. 100. 100." />
<mesh filename="package://atlas/meshes/r_uleg.dae" scale="100. 100. 100." />
~~~

**Comment by osrf-migration** [comment]
**Original comment by Jesper Smith (Bitbucket: [jespersmith](https://bitbucket.org/%7B61d9fe7d-e554-408c-99fb-faefa9c96114%7D/)).**

----------------------------------------
I disabled the scaling, but I'm seeing weird translations when importing the model in our user interface and sim. I've attached a screenshot, it seems that the meshes that are offset match the ones in centimetre resolution.

Is it possible to export these meshes in the same way the other meshes are saved?

Note that the previous model imported find in our software.

**Comment by osrf-migration** [comment]
**Original comment by Jesper Smith (Bitbucket: [jespersmith](https://bitbucket.org/%7B61d9fe7d-e554-408c-99fb-faefa9c96114%7D/)).**

----------------------------------------

* set attachment to "NewModelImport.png"

**Comment by osrf-migration** [comment]
**Original comment by Steve Peters (Bitbucket: [Steven Peters](https://bitbucket.org/%7B2ccfed09-18b8-4921-8d58-15ef01092802%7D/), GitHub: [scpeters](https://github.com/scpeters)).**

----------------------------------------
Here's a patch for the pelvis that still appears fine in gazebo. Does it help your parser?
~~~
diff -r e4a9d846a87fa7ec41ab7408e635b335c0898bcb ros/atlas_description/meshes/pelvis.dae
--- a/ros/atlas_description/meshes/pelvis.dae	Thu May 16 14:06:25 2013 +0200
+++ b/ros/atlas_description/meshes/pelvis.dae	Thu May 16 14:51:02 2013 -0700
@@ -12,7 +12,7 @@
     <revision></revision>
     <subject></subject>
     <title></title>
-    <unit meter="1.0" name="centimeter"></unit>
+    <unit meter="100.0" name="meter"></unit>
     <up_axis>Z_UP</up_axis>
   </asset>
   <library_images>
diff -r e4a9d846a87fa7ec41ab7408e635b335c0898bcb ros/atlas_description/urdf/atlas_simple_shapes.urdf
--- a/ros/atlas_description/urdf/atlas_simple_shapes.urdf	Thu May 16 14:06:25 2013 +0200
+++ b/ros/atlas_description/urdf/atlas_simple_shapes.urdf	Thu May 16 14:51:02 2013 -0700
@@ -381,7 +381,7 @@
         <visual>
             <origin xyz="0 0 0" rpy="0 -0 0" />
             <geometry>
-                <mesh filename="package://atlas/meshes/pelvis.dae" scale="100. 100. 100." />
+                <mesh filename="package://atlas/meshes/pelvis.dae" scale="1 1 1" />
             </geometry>
         </visual>
         <collision group="default">
~~~

**Comment by osrf-migration** [comment]
**Original comment by Jesper Smith (Bitbucket: [jespersmith](https://bitbucket.org/%7B61d9fe7d-e554-408c-99fb-faefa9c96114%7D/)).**

----------------------------------------
Both setting 
+    <unit meter="100.0" name="meter"></unit> (and changing the URDF)
or
+    <unit meter="0.01" name="centimeter"></unit>

work for the scale. However, I still get the offsets which I cannot figure out where they come from. I've drawn all meshes on a grid and the offending meshes are shifted with respect to 0,0,0. I'm not sure if the importer we're using is missing something or there is something else going on though.

**Comment by osrf-migration** [comment]
**Original comment by Jesper Smith (Bitbucket: [jespersmith](https://bitbucket.org/%7B61d9fe7d-e554-408c-99fb-faefa9c96114%7D/)).**

----------------------------------------

* changed state from "new" to "resolved"

**Comment by osrf-migration** [comment]
**Original comment by Jesper Smith (Bitbucket: [jespersmith](https://bitbucket.org/%7B61d9fe7d-e554-408c-99fb-faefa9c96114%7D/)).**

----------------------------------------

* changed state from "resolved" to "invalid"

My bad, there is a scale in the transformation matrix in the DAE files:

<matrix sid="matrix">0.010000 0.000000 0.000000 0.000000 0.000000 0.010000 0.000000 0.000000 0.000000 0.000000 0.010000 0.000000 0.000000 0.000000 0.000000 1.000000</matrix>

The importer we're using didn't handle that correctly. I've fixed that, so the scales are correct. I still have the weird offsets, looking into that. If necessary, I'll make a new bugreport.

**Comment by osrf-migration** [comment]
**Original comment by Jesper Smith (Bitbucket: [jespersmith](https://bitbucket.org/%7B61d9fe7d-e554-408c-99fb-faefa9c96114%7D/)).**

----------------------------------------

* changed state from "invalid" to "open"

I could not fix the third party DAE loader we're using without a lot of effort to use scales correctly (it applied the scale to the translations of the next mesh transforms in a DAE file), so I changed the DAE files to be in meter coordinates and they look good in both Gazebo and our UI now.

I've attached the changed meshes and would like to request they be included in the official drcsim before the freeze, if possible.

**Comment by osrf-migration** [comment]
**Original comment by Jesper Smith (Bitbucket: [jespersmith](https://bitbucket.org/%7B61d9fe7d-e554-408c-99fb-faefa9c96114%7D/)).**

----------------------------------------

* set attachment to "models.zip"

**Comment by osrf-migration** [comment]
**Original comment by Steve Peters (Bitbucket: [Steven Peters](https://bitbucket.org/%7B2ccfed09-18b8-4921-8d58-15ef01092802%7D/), GitHub: [scpeters](https://github.com/scpeters)).**

----------------------------------------
Did you also change the urdf file? Those models look huge to me.

**Comment by osrf-migration** [comment]
**Original comment by Jesper Smith (Bitbucket: [jespersmith](https://bitbucket.org/%7B61d9fe7d-e554-408c-99fb-faefa9c96114%7D/)).**

----------------------------------------
Yeah, changed the scale to 1.0 for hese models

**Comment by osrf-migration** [comment]
**Original comment by Jesper Smith (Bitbucket: [jespersmith](https://bitbucket.org/%7B61d9fe7d-e554-408c-99fb-faefa9c96114%7D/)).**

----------------------------------------

* set attachment to "atlas_simple_shapes.urdf"

Scales in URDF file

**Comment by osrf-migration** [comment]
**Original comment by Steve Peters (Bitbucket: [Steven Peters](https://bitbucket.org/%7B2ccfed09-18b8-4921-8d58-15ef01092802%7D/), GitHub: [scpeters](https://github.com/scpeters)).**

----------------------------------------
It looks good to me and appears to make things more consistent. I've created [pull request #303](https://osrf-migration.github.io/drcsim-gh-pages/#!/osrf/drcsim/pull-requests/303).

**Comment by osrf-migration** [comment]
**Original comment by Jesper Smith (Bitbucket: [jespersmith](https://bitbucket.org/%7B61d9fe7d-e554-408c-99fb-faefa9c96114%7D/)).**

----------------------------------------

* changed state from "open" to "resolved"

Resolved by [pull request #303](https://osrf-migration.github.io/drcsim-gh-pages/#!/osrf/drcsim/pull-requests/303)
